projects/angular/src/data/datagrid/datagrid-row.spec.ts

```diff
@@ -543,6 +544,115 @@ export default function (): void {
+        expect(button.querySelector('cds-icon[shape^=angle-double][direction^=right]')).not.toBeNull();
+
+        button.click();
+        tick();
+        context.detectChanges();
+
+        const detailBody = document.querySelector('div.clr-dg-detail-body-wrapper') as HTMLElement;
+        expect(button.querySelector('cds-icon[shape^=angle-double][direction^=left]')).not.toBeNull();
+        expect(detailBody.innerText).toBe('' + context.testComponent.items[0].id);
+
+        button.click();
+        tick();
+        context.detectChanges();
+
+        expect(button.querySelector('cds-icon[shape^=angle-double][direction^=right]')).not.toBeNull();
+      }));
+
+      it('try open the disabled details', fakeAsync(function () {
+        context.testComponent.disabledIndex = 0;
+        tick();
+        context.detectChanges();
+
+        const button = context.clarityElement.querySelector('button.datagrid-detail-caret-button[disabled]');
+        expect(button).not.toBeNull();
+        expect(button.querySelector('cds-icon[shape^=angle-double][direction^=right]')).not.toBeNull();
+
+        button.click();
+        tick();
+        context.detectChanges();
+
+        expect(button).not.toBeNull();
+        expect(button.querySelector('cds-icon[shape^=angle-double][direction^=right]')).not.toBeNull();
+      }));
+
+      it('disabled prestate details can opened', fakeAsync(function () {
+        context.testComponent.disabledIndex = 0;
+        tick();
+        context.detectChanges();
+
+        context.testComponent.preState = context.testComponent.items[context.testComponent.disabledIndex];
+        tick();
+        context.detectChanges();
+
+        const button = context.clarityElement.querySelector('button.datagrid-detail-caret-button[disabled]');
+        expect(button).not.toBeNull();
+        expect(button.querySelector('cds-icon[shape^=angle-double][direction^=left]')).not.toBeNull();
+      }));
+
+      it('try open the hidden detail button', fakeAsync(function () {
+        context.testComponent.hiddenIndex = 0;
+        tick();
+        context.detectChanges();
+        tick();
+        context.detectChanges();
+
+        const buttons = context.clarityElement.querySelectorAll('button.datagrid-detail-caret-button');
+        expect(buttons.length).toBe(1);
+      }));
+
+      it('hidden detail button prestate details are opened', fakeAsync(function () {
+        context.testComponent.hiddenIndex = 0;
+        tick();
+        context.detectChanges();
+
+        context.testComponent.preState = context.testComponent.items[context.testComponent.hiddenIndex];
+        tick();
+        context.detectChanges();
+
+        const buttons = context.clarityElement.querySelectorAll('button.datagrid-detail-caret-button');
+        expect(buttons.length).toBe(1);
+
+        expect(context.clarityElement.querySelector('clr-dg-detail')).not.toBeNull();
+      }));
+    });
```

Comment:
Not sure if we need that for the hidden functionality. We need to just verify if it's properly hiding the button and thats it.

Reply:
removed in commit 0ad320b